subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublication.java

```diff
@@ -528,9 +528,10 @@ public void whenPublishableArtifactRemoved(Action<? super IvyArtifact> action) {
 
     @Override
     public IvyArtifact addDerivedArtifact(IvyArtifact originalArtifact, DerivedArtifact fileProvider) {
-        if (originalArtifact == gradleModuleDescriptorArtifact) {
-            return null;
-        }
+        fileProvider = originalArtifact == gradleModuleDescriptorArtifact
```

Comment:
💄 I'd avoid changing the original parameter here. I'd introduce a new variable, say, `effectiveFileProvider` and use it below.